geopandas/sindex.py

```diff
@@ -70,6 +91,150 @@ def __init__(self, geometry):
                 super().__init__()
 
+            # store reference to geometries for predicate queries
+            self._geometries = geometry.geometry.values
+            # create a prepared geometry cache
+            self._prepared_geometries = np.array(
+                [None] * self._geometries.size, dtype=object
+            )
+
+        def query(self, geometry, predicate=None, sort=False):
+            """Compatibility layer for pygeos.query.
+
+            This is not a vectorized function, if speed is important,
+            please use PyGEOS.
+
+            Parameters
+            ----------
+            geometry : shapely geometry
+                A single shapely geometry to query against the spatial index.
+            predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses', 'touches'}, optional
+                If predicate is provided, the input geometry is tested using the predicate function against each
+                item in the tree whose extent intersects the envelope of the input geometry:
+                predicate(input_geometry, tree_geometry).  If possible, prepared geometries are used to help
+                speed up the predicate operation.
+            sort : bool, default False
+                If True, the results will be sorted in ascending order. If False, results are
+                often sorted but there is no guarantee.
+
+            Returns
+            -------
+            matches : ndarray of shape (n_results, )
+                Integer indices for matching geometries from the spatial index.
+            """  # noqa: E501
+
+            # handle invalid predicates
+            if predicate not in self.valid_query_predicates:
+                raise ValueError(
+                    "Got `predicate` = `{}`, `predicate` must be one of {}".format(
+                        predicate, self.valid_query_predicates
+                    )
+                )
+
+            # handle empty / invalid geometries
+            if geometry is None:
+                # return an empty integer array, similar to pygeys.STRtree.query.
+                return np.array([], dtype=np.intp)
+
+            if not isinstance(geometry, BaseGeometry):
+                raise TypeError(
+                    "Got `geometry` of type `{}`, `geometry` must be ".format(
+                        type(geometry)
+                    )
+                    + "a shapely geometry."
+                )
+
+            if geometry.is_empty:
+                return np.array([], dtype=np.intp)
+
+            # query tree
+            bounds = geometry.bounds  # rtree operates on bounds
+            tree_idx = list(self.intersection(bounds, objects=False))
+
+            if not tree_idx:
+                return np.array([], dtype=np.intp)
+
+            # Check predicate
+            # When possible, we want to use cached prepared geometries
+            # from the tree to compare tree_geom.predicate(input_geom)
```

Comment:
The other way around, right? (`input_geom.predicate(tree_geom)`) 
At least, that's what I think pygeos does? (from its docstring: `predicate(geometry, tree_geometry)`), and also what you put in a comment below